From 5c325fe342c26f31df495ec478a1d427270e8e42 Mon Sep 17 00:00:00 2001
From: cgahr <constantin.gahr+github@pm.me>
Date: Fri, 3 Nov 2023 22:21:54 +0100
Subject: [PATCH 026/188] replace kdl tree-sitter to fix highlighting (#8652)

* replace kdl tree-sitter

* kdl: adopt highlights for new tree-sitter

* kdl: add indent queries

* kdl: add textobjects

* kdl: improve syntax highlighting

* kdl: update lang-support

* kdl: make indents more concise

---------

Co-authored-by: Constantin Gahr <constantin.gahr@ipp.mpg.de>
---
 book/src/generated/lang-support.md  |  2 +-
 languages.toml                      |  2 +-
 runtime/queries/kdl/highlights.scm  |  8 ++++----
 runtime/queries/kdl/indents.scm     |  3 +++
 runtime/queries/kdl/textobjects.scm | 27 +++++++++++++++++++++++++++
 5 files changed, 36 insertions(+), 6 deletions(-)
 create mode 100644 runtime/queries/kdl/indents.scm
 create mode 100644 runtime/queries/kdl/textobjects.scm

--- a/book/src/generated/lang-support.md
+++ b/book/src/generated/lang-support.md
@@ -81,7 +81,7 @@
 | jsx | ✓ | ✓ | ✓ | `typescript-language-server` |
 | julia | ✓ | ✓ | ✓ | `julia` |
 | just | ✓ | ✓ | ✓ |  |
-| kdl | ✓ |  |  |  |
+| kdl | ✓ | ✓ | ✓ |  |
 | kotlin | ✓ |  |  | `kotlin-language-server` |
 | latex | ✓ | ✓ |  | `texlab` |
 | lean | ✓ |  |  | `lean` |
--- a/languages.toml
+++ b/languages.toml
@@ -2340,7 +2340,7 @@ injection-regex = "kdl"
 
 [[grammar]]
 name = "kdl"
-source = { git = "https://github.com/Unoqwy/tree-sitter-kdl", rev = "e1cd292c6d15df6610484e1d4b5c987ecad52373" }
+source = { git = "https://github.com/amaanq/tree-sitter-kdl", rev = "3ca569b9f9af43593c24f9e7a21f02f43a13bb88" }
 
 [[language]]
 name = "xml"
--- a/runtime/queries/kdl/highlights.scm
+++ b/runtime/queries/kdl/highlights.scm
@@ -1,12 +1,12 @@
-(comment) @comment
 (single_line_comment) @comment
+(multi_line_comment) @comment
 
 (node
-    name: (identifier) @function)
+    (identifier) @variable)
+
 (prop (identifier) @attribute)
-(type) @type
 
-(bare_identifier) @variable.other.member
+(type (_) @type) @punctuation.bracket
 
 (keyword) @keyword
 
--- /dev/null
+++ b/runtime/queries/kdl/indents.scm
@@ -0,0 +1,3 @@
+(node_children) @indent
+
+ "}" @outdent
--- /dev/null
+++ b/runtime/queries/kdl/textobjects.scm
@@ -0,0 +1,27 @@
+(type (_) @test.inside) @test.around
+
+(node
+	children: (node_children)? @class.inside) @class.around
+
+(node
+	children: (node_children)? @function.inside) @function.around
+
+(node (identifier) @function.movement)
+
+[
+	(single_line_comment)
+	(multi_line_comment)
+] @comment.inside
+
+[
+	(single_line_comment)+
+	(multi_line_comment)+
+] @comment.around
+
+[
+	(prop)
+	(value)
+] @parameter.inside
+
+(value (type) ? (_) @parameter.inside @parameter.movement . ) @parameter.around
+
